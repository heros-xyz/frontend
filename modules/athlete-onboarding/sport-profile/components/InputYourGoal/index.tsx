import { Box, Input, Text } from "@chakra-ui/react";
import * as Yup from "yup";
import { useFormik } from "formik";
import HerosOnboardingWrapper from "@/components/ui/HerosOnboardingWrapper";
import { YourGoalIcon } from "@/components/svg/YourGoalIcon";
import ErrorMessage from "@/components/common/ErrorMessage";
interface IProps {
  onSubmit: (value: object) => void;
}

const InputYourGoal: React.FC<IProps> = ({ onSubmit }) => {
  const validationSchema = Yup.object().shape({
    goal: Yup.string()
      .max(500, "Your Goal cannot exceed 500 characters.")
      .required("This is a required field"),
  });

  const formik = useFormik({
    initialValues: {
      goal: "",
    },
    validationSchema: validationSchema,
    onSubmit: (values) => {
      onSubmit(values);
    },
  });
  return (
    <HerosOnboardingWrapper
      Icon={<YourGoalIcon w="full" h="full" />}
      textButton="Submit"
      title="SPORT PROFILE"
      onSubmit={formik.handleSubmit}
    >
      <Box mb={4}>
        <Box mb={2}>
          <Box>
            <Text as="span" fontWeight="600">
              Enter Your Goal (500 characters max)
            </Text>
            <Text as="span" color={"red"} ml={1}>
              *
            </Text>
          </Box>
          <Text fontSize={"xs"} mt={2}>
            Make sure you paint a compelling picture of how they can join you on
            this journey.
          </Text>
        </Box>
        <Input
          id="goal"
          variant={"flushed"}
          value={formik.values.goal}
          borderColor="primary"
          placeholder="Your Goal"
          onChange={formik.handleChange}
          isInvalid={Boolean(formik.errors.goal && formik.touched.goal)}
        />
        <ErrorMessage
          condition={formik.errors.goal && formik.touched.goal}
          errorMessage={formik.errors.goal}
        />
      </Box>
    </HerosOnboardingWrapper>
  );
};

export default InputYourGoal;
