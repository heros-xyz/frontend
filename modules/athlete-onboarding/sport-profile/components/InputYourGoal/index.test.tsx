import { render } from "@testing-library/react";
import InputYourGoal from ".";

it("renders add Input Your Goal unchanged", () => {
  const { container } = render(<InputYourGoal />);
  expect(container).toMatchSnapshot();
});
