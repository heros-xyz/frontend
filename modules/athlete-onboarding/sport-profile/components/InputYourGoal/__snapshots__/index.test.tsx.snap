// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders add Input Your Goal unchanged 1`] = `
<div>
  <div
    class="css-1jvt6v8"
  >
    <div
      class="css-1kmmdcq"
    >
      Sport profile
    </div>
    <div
      class="chakra-container css-17f3xev"
    >
      <div
        class="css-1ispe57"
      >
        <div
          class="css-l29o6j"
        >
          <div
            class="css-1jz6iun"
          >
            <svg
              class="chakra-icon css-1e009k1"
              focusable="false"
              viewBox="0 0 151 150"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="75.5"
                cy="75"
                fill="currentColor"
                r="74.5"
                stroke="#DDDDDD"
              />
              <g
                clip-path="url(#clip0_1504_20084)"
              >
                <path
                  clipRule="evenodd"
                  d="M103.561 60.8117C93.0007 40.8258 69.2407 31.9165 50.6407 40.9462C31.9207 50.0964 25.4407 73.6941 35.8807 93.6799C46.4407 113.666 70.0807 122.575 88.8007 113.545C107.521 104.395 114.001 80.7975 103.561 60.8117Z"
                  fill="#FFFBEC"
                  fillRule="evenodd"
                />
                <path
                  d="M103.561 60.8117C93.0007 40.8258 69.2407 31.9165 50.6407 40.9462C31.9207 50.0964 25.4407 73.6941 35.8807 93.6799C46.4407 113.666 70.0807 122.575 88.8007 113.545C107.521 104.395 114.001 80.7975 103.561 60.8117Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M103.561 60.8117C93.0007 40.8258 69.2407 31.9165 50.6407 40.9462C31.9207 50.0964 25.4407 73.6941 35.8807 93.6799C46.4407 113.666 70.0807 122.575 88.8007 113.545C107.521 104.395 114.001 80.7975 103.561 60.8117Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clipRule="evenodd"
                  d="M102.721 47.3272C118.321 63.8215 119.041 88.9844 104.281 103.552C89.5208 118.12 65.0408 116.676 49.4408 100.181C33.8408 83.687 33.1208 58.5241 47.8808 43.9561C62.5208 29.3881 87.1208 30.8329 102.721 47.3272Z"
                  fill="white"
                  fillRule="evenodd"
                />
                <path
                  d="M102.721 47.3272C118.321 63.8215 119.041 88.9844 104.281 103.552C89.5208 118.12 65.0408 116.676 49.4408 100.181C33.8408 83.687 33.1208 58.5241 47.8808 43.9561C62.5208 29.3881 87.1208 30.8329 102.721 47.3272Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M102.721 47.3272C118.321 63.8215 119.041 88.9844 104.281 103.552C89.5208 118.12 65.0408 116.676 49.4408 100.181C33.8408 83.687 33.1208 58.5241 47.8808 43.9561C62.5208 29.3881 87.1208 30.8329 102.721 47.3272Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clipRule="evenodd"
                  d="M100.081 49.7351C114.121 64.6642 114.841 87.4192 101.521 100.542C88.2011 113.786 66.0011 112.341 51.9611 97.5325C37.8011 82.6033 37.2011 59.8484 50.5211 46.6047C63.8411 33.4815 86.0411 34.8059 100.081 49.7351Z"
                  fill="white"
                  fillRule="evenodd"
                />
                <path
                  d="M100.081 49.7351C114.121 64.6642 114.841 87.4192 101.521 100.542C88.2011 113.786 66.0011 112.341 51.9611 97.5325C37.8011 82.6033 37.2011 59.8484 50.5211 46.6047C63.8411 33.4815 86.0411 34.8059 100.081 49.7351Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M100.081 49.7351C114.121 64.6642 114.841 87.4192 101.521 100.542C88.2011 113.786 66.0011 112.341 51.9611 97.5325C37.8011 82.6033 37.2011 59.8484 50.5211 46.6047C63.8411 33.4815 86.0411 34.8059 100.081 49.7351Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clipRule="evenodd"
                  d="M91.6799 58.0426C100.8 67.6743 101.28 82.4831 92.6399 91.0312C83.9999 99.5794 69.5999 98.7366 60.3599 89.1049C51.2399 79.4731 50.7599 64.6644 59.3999 56.1162C68.0399 47.568 82.5599 48.4108 91.6799 58.0426Z"
                  fill="white"
                  fillRule="evenodd"
                />
                <path
                  d="M91.6799 58.0426C100.8 67.6743 101.28 82.4831 92.6399 91.0312C83.9999 99.5794 69.5999 98.7366 60.3599 89.1049C51.2399 79.4731 50.7599 64.6644 59.3999 56.1162C68.0399 47.568 82.5599 48.4108 91.6799 58.0426Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M91.6799 58.0426C100.8 67.6743 101.28 82.4831 92.6399 91.0312C83.9999 99.5794 69.5999 98.7366 60.3599 89.1049C51.2399 79.4731 50.7599 64.6644 59.3999 56.1162C68.0399 47.568 82.5599 48.4108 91.6799 58.0426Z"
                  stroke="#ED2024"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clipRule="evenodd"
                  d="M83.401 66.3498C87.721 70.9249 87.961 77.9079 83.881 81.881C79.801 85.8541 73.081 85.4929 68.761 80.9178C64.441 76.3427 64.201 69.3597 68.281 65.3866C72.241 61.2931 79.081 61.7747 83.401 66.3498Z"
                  fill="#ED2024"
                  fillRule="evenodd"
                />
                <path
                  d="M76.2012 73.6941L120.721 46.0029"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clipRule="evenodd"
                  d="M100.441 58.8852C101.521 57.1996 102.361 53.4673 101.881 48.2903C101.881 47.8087 101.761 47.3271 101.761 46.8455L117.721 37.4546C119.041 40.2237 118.921 47.2067 117.841 47.4475L100.441 58.8852Z"
                  fill="#D7E037"
                  fillRule="evenodd"
                />
                <path
                  d="M100.441 58.8852C101.521 57.1996 102.361 53.4673 101.881 48.2903C101.881 47.8087 101.761 47.3271 101.761 46.8455L117.721 37.4546C119.041 40.2237 118.921 47.2067 117.841 47.4475L100.441 58.8852Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M100.441 58.8852C101.521 57.1996 102.361 53.4673 101.881 48.2903C101.881 47.8087 101.761 47.3271 101.761 46.8455L117.721 37.4546C119.041 40.2237 118.921 47.2067 117.841 47.4475L100.441 58.8852Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  clipRule="evenodd"
                  d="M100.441 58.6445C102.481 58.2833 106.321 59.1261 111.121 61.534C111.601 61.7748 111.961 62.0156 112.441 62.2564L127.681 51.9023C125.641 49.6148 118.801 46.966 118.201 47.8088L100.441 58.6445Z"
                  fill="#D7E037"
                  fillRule="evenodd"
                />
                <path
                  d="M100.441 58.6445C102.481 58.2833 106.321 59.1261 111.121 61.534C111.601 61.7748 111.961 62.0156 112.441 62.2564L127.681 51.9023C125.641 49.6148 118.801 46.966 118.201 47.8088L100.441 58.6445Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M100.441 58.6445C102.481 58.2833 106.321 59.1261 111.121 61.534C111.601 61.7748 111.961 62.0156 112.441 62.2564L127.681 51.9023C125.641 49.6148 118.801 46.966 118.201 47.8088L100.441 58.6445Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_1504_20084"
                >
                  <rect
                    fill="white"
                    height="85"
                    transform="translate(30 33)"
                    width="99"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
        <div
          class="css-14s5ckc"
        >
          <div
            class="css-1s2o82u"
          >
            <div
              class="css-1uoge8t"
            >
              <div
                class="css-1itv5e3"
              >
                <div
                  class="css-wlea3r"
                >
                  <div
                    class="css-0"
                  >
                    <span
                      class="chakra-text css-1jjm4rf"
                    >
                      Enter Your Goal (500 characters max)
                    </span>
                    <span
                      class="chakra-text css-684fo5"
                    >
                      *
                    </span>
                  </div>
                  <p
                    class="chakra-text css-ucpxtz"
                  >
                    Make sure you paint a compelling picture of how they can join you on this journey.
                  </p>
                </div>
                <input
                  class="chakra-input css-1oz8u65"
                  id="goal"
                  placeholder="Your Goal"
                  value=""
                />
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  Submit
                  <span
                    class="css-9n3bbc"
                  />
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            type="button"
          >
            Submit
            <span
              class="css-9n3bbc"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
